# HG changeset patch
# User alta88@fixall.com
# Date 1750335418 -7200
# Parent  e3b8acf2ccc7d9e812619bcca7d60f33420012e1
Feed Enhancements - make what is a feed message portable:
- Ensure FeedMsg flag is carried over to imap copy
- Add a Content-Description header with feed type and url to the message

diff --git a/mailnews/extensions/newsblog/FeedItem.sys.mjs b/mailnews/extensions/newsblog/FeedItem.sys.mjs
--- a/mailnews/extensions/newsblog/FeedItem.sys.mjs
+++ b/mailnews/extensions/newsblog/FeedItem.sys.mjs
@@ -313,8 +313,10 @@ FeedItem.prototype = {
       keywordsStr +
       "Content-Transfer-Encoding: 8bit\n" +
       "Content-Base: " +
       this.mURL +
+      "\n" +
+      `Content-Description: feed-message="${this.feed.mFeedType}"; feed-url="${this.feed.url}"` +
       "\n";
 
     if (this.enclosures.length) {
       const boundaryID = source.length;
diff --git a/mailnews/imap/src/nsImapMailFolder.cpp b/mailnews/imap/src/nsImapMailFolder.cpp
--- a/mailnews/imap/src/nsImapMailFolder.cpp
+++ b/mailnews/imap/src/nsImapMailFolder.cpp
@@ -6921,8 +6921,13 @@ void nsImapMailFolder::SetPendingAttribu
     if (flags & nsMsgMessageFlags::HasRe) {
       storeFlags = nsMsgMessageFlags::HasRe;
       mDatabase->SetUint32AttributeOnPendingHdr(msgDBHdr, "flags", storeFlags);
     }
+    // And the FeedMsg flag.
+    if (flags & nsMsgMessageFlags::FeedMsg) {
+      mDatabase->SetUint32AttributeOnPendingHdr(
+          msgDBHdr, "flags", storeFlags | nsMsgMessageFlags::FeedMsg);
+    }
 
     uint32_t messageSize;
     msgDBHdr->GetOfflineMessageSize(&messageSize);
     if (messageSize) {
